nexosim/src/simulation.rs

```diff
@@ -739,6 +738,130 @@ pub struct DeadlockInfo {
     pub mailbox_size: usize,
 }
 
+/// An error returned upon failure during simulation state store procedure.
+#[non_exhaustive]
+#[derive(Debug)]
+pub enum SaveError {
+    #[cfg(feature = "server")]
+    /// Cbor serialization of the simulation's config has failed.
+    ConfigSerializationError(ciborium::ser::Error<std::io::Error>),
```

Comment:
I'm afraid having feature-dependent variants could lead to fragile user code, especially if the user is making a library and does not control the `server` feature.

We could avoid the `cfg` and the dependency on `ciborium` by using instead a plain `std::io::Error`:

- a `ciborium::ser::Io(err)`  would be mapped to the inner `err`,
- a `ciborium::ser::Value(err_string) would be mapped to `std::io::Error`, using `std::io::Error::new(ErrorKind::Other, err_string)`;